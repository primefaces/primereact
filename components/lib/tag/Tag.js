import * as React from 'react';
import { classNames, IconUtils, ObjectUtils } from '../utils/Utils';

export const Tag = React.forwardRef((props, ref) => {
    const otherProps = ObjectUtils.findDiffKeys(props, Tag.defaultProps);
    const className = classNames('p-tag p-component', {
        [`p-tag-${props.severity}`]: props.severity !== null,
        'p-tag-rounded': props.rounded
    }, props.className);
    const icon = IconUtils.getJSXIcon(props.icon, { className: 'p-tag-icon' }, { props });

    return (
        <span className={className} style={props.style} {...otherProps}>
            {icon}
            <span className="p-tag-value">{props.value}</span>
            <span>{props.children}</span>
        </span>
    )
});

Tag.displayName = 'Tag';
Tag.defaultProps = {
    __TYPE: 'Tag',
    value: null,
    severity: null,
    rounded: false,
    icon: null,
    style: null,
    className: null
}
