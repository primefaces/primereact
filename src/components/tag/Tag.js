import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { classNames } from '../utils/Utils';

export class Tag extends Component {

    static defaultProps = {
        value: null,
        severity: null,
        rounded: false,
        icon: null,
        style: null,
        className: null
    }

    static propTypes = {
        value: PropTypes.any,
        severity: PropTypes.string,
        rounded: PropTypes.bool,
        icon: PropTypes.string,
        style: PropTypes.object,
        className: PropTypes.string
    };

    render() {
        const tagClassName = classNames('p-tag p-component', {
            'p-tag-info': this.props.severity === 'info',
            'p-tag-success': this.props.severity === 'success',
            'p-tag-warning': this.props.severity === 'warning',
            'p-tag-danger': this.props.severity === 'danger',
            'p-tag-rounded': this.props.rounded
        }, this.props.className);

        const iconClass = classNames('p-tag-icon', this.props.icon);

        return (
            <span className={tagClassName} style={this.props.style}>
                {this.props.icon && <span className={iconClass}></span>}
                <span className="p-tag-value">{this.props.value}</span>
                <span>{this.props.children}</span>
            </span>
        );
    }
}
