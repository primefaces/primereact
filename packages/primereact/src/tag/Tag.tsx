'use client';
import { Component, withComponent } from '@primereact/core/component';
import { useTag } from '@primereact/headless/tag';
import { styles } from '@primereact/styles/tag';
import { mergeProps } from '@primeuix/utils';
import * as React from 'react';
import { defaultProps } from './Tag.props';
import { TagIcon } from './icon';
import { TagLabel } from './label';

export const Tag = withComponent({
    defaultProps,
    styles,
    setup: (instance) => {
        const tag = useTag(instance.inProps);

        return tag;
    },
    render: ({
        id,
        props,
        state,
        ptmi,
        ptm,
        cx,
        // element refs
        elementRef
        // methods
    }) => {
        const rootProps = mergeProps(
            {
                id,
                className: cx('root')
            },
            ptmi('root')
        );

        return state.isRemoved ? null : (
            <Component as={props.as || 'span'} {...rootProps} ref={elementRef}>
                {props.children}
            </Component>
        );
    },
    components: {
        Icon: TagIcon,
        Label: TagLabel
    }
});
