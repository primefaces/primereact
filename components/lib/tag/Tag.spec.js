import '@testing-library/jest-dom';
import { Tag } from './Tag';

import { snapshot } from '../../test';

describe('Tag', () => {
    snapshot(<Tag />, 'default');
    snapshot(<Tag value={`jest`} />, 'value');
    snapshot(<Tag severity="success" />, 'sevrity success');
    snapshot(<Tag severity="info" />, 'sevrity info');
    snapshot(<Tag severity="warning" />, 'sevrity warning');
    snapshot(<Tag severity="danger" />, 'sevrity danger');
    snapshot(<Tag rounded />, 'rounded');
    snapshot(<Tag icon="pi pi-check" />, 'icon');
    snapshot(
        <Tag>
            <div>hello</div>
        </Tag>,
        'templating'
    );
});
