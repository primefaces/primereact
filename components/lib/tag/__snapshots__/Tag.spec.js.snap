// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag default 1`] = `
<div>
  <span
    class="p-tag p-component"
  >
    <span
      class="p-tag-value"
    />
    <span />
  </span>
</div>
`;

exports[`Tag icon 1`] = `
<div>
  <span
    class="p-tag p-component"
  >
    <span
      class="p-tag-icon pi pi-check"
    />
    <span
      class="p-tag-value"
    />
    <span />
  </span>
</div>
`;

exports[`Tag rounded 1`] = `
<div>
  <span
    class="p-tag p-component p-tag-rounded"
  >
    <span
      class="p-tag-value"
    />
    <span />
  </span>
</div>
`;

exports[`Tag sevrity danger 1`] = `
<div>
  <span
    class="p-tag p-component p-tag-danger"
  >
    <span
      class="p-tag-value"
    />
    <span />
  </span>
</div>
`;

exports[`Tag sevrity info 1`] = `
<div>
  <span
    class="p-tag p-component p-tag-info"
  >
    <span
      class="p-tag-value"
    />
    <span />
  </span>
</div>
`;

exports[`Tag sevrity success 1`] = `
<div>
  <span
    class="p-tag p-component p-tag-success"
  >
    <span
      class="p-tag-value"
    />
    <span />
  </span>
</div>
`;

exports[`Tag sevrity warning 1`] = `
<div>
  <span
    class="p-tag p-component p-tag-warning"
  >
    <span
      class="p-tag-value"
    />
    <span />
  </span>
</div>
`;

exports[`Tag templating 1`] = `
<div>
  <span
    class="p-tag p-component"
  >
    <span
      class="p-tag-value"
    />
    <span>
      <div>
        hello
      </div>
    </span>
  </span>
</div>
`;

exports[`Tag value 1`] = `
<div>
  <span
    class="p-tag p-component"
  >
    <span
      class="p-tag-value"
    >
      jest
    </span>
    <span />
  </span>
</div>
`;
